src/main/java/com/minecolonies/coremod/quests/objectives/DeliveryObjectiveTemplateTemplate.java

```diff
@@ -137,4 +139,80 @@ public DialogueElement getReadyDialogueTree()
     {
         return readyDialogueElement;
     }
+
+    @Override
+    public @NotNull IObjectiveInstance createObjectiveInstance()
+    {
+        return new DeliveryProgressInstance(this);
+    }
+
+    /**
+     * Progress data of this objective.
+     */
+    public static class DeliveryProgressInstance implements IDialogueObjectInstance
```

Comment:
I don't agree with adding an instance to all objectives because delivery and dialogue move on. There is no need to store data for them.